system-addon/lib/ShortURL.jsm

```diff
@@ -20,7 +40,7 @@ this.shortURL = function shortURL(link) {
     return "";
   }
   const {eTLD} = link;
-  const hostname = (link.hostname || new URL(link.url).hostname).replace(/^www\./i, "");
+  const hostname = handleIDNHost(link.hostname || new URL(link.url).hostname).replace(/^www\./i, "");
```

Comment:
Did you intend to strip the `www.` before or after `handleIDNHost`? Right now it's after, and it seems like it doesn't matter here. I suppose one way to be clearer about where you meant is to just split up this line to be
```js
const asciiHost = link.hostname…
const hostname = handleIDN…
```